### PR TITLE
[11.0][MIG] purchase_order_variant_mgmt

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: purchase_order_variant_mgmt/tests/test_purchase_order_variant_mgmt.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests import common
+
+
+class TestPurchaseOrderVariantMgmt(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.partner = cls.env['res.partner'].create({'name': 'Test partner'})
```
Comment (simahawk) [inline purchase_order_variant_mgmt/tests/test_purchase_order_variant_mgmt.py:11]: pls use tracking_disable on the whole env

File: purchase_order_variant_mgmt/wizard/purchase_manage_variant.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        string="Template",
+        required=True,
+    )
+    # This is a many2many because Odoo fails to fill one2many in onchanges
+    variant_line_ids = fields.Many2many(
+        comodel_name='purchase.manage.variant.line',
+        string="Variant Lines",
+    )
+
+    # HACK: https://github.com/OCA/server-tools/pull/492#issuecomment-237594285
```
Comment (simahawk) [inline purchase_order_variant_mgmt/wizard/purchase_manage_variant.py:30]: is this still needed?

**Review by simahawk — CHANGES_REQUESTED** [review]
LG, just minor things

File: purchase_order_variant_mgmt/tests/test_purchase_order_variant_mgmt.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        cls.company = cls.env['res.company'].create({'name': 'Test company'})
+        cls.attribute1 = cls.env['product.attribute'].create({
```
Comment (nikul-serpentcs) [inline purchase_order_variant_mgmt/tests/test_purchase_order_variant_mgmt.py:13]: @arkostyuk IMO If use `cls.env['product.attribute']`  multiple time then created variable for globle.

File: purchase_order_variant_mgmt/views/purchase_order_view.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <record id="purchase_order_form" model="ir.ui.view">
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='order_line']" position="before">
```
Comment (nikul-serpentcs) [inline purchase_order_variant_mgmt/views/purchase_order_view.xml:10]: here Direct use `<field>`, remove `xpath`

File: purchase_order_variant_mgmt/views/purchase_order_view.xml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                <button name="%(action_purchase_manage_variant)d"
+                        type="action"
+                        string="Modify Variants"
+                        icon="fa-th"
+                        class="oe_edit_only"
+                        context="{'default_product_tmpl_id': product_tmpl_id_purchase_order_variant_mgmt}"
+                        attrs="{'invisible': ['|', ('state_purchase_order_variant_mgmt', 'not in', ('draft', 'sent')), ('product_attribute_value_ids', '=', [])]}"
+                />
+            </xpath>
+            <xpath expr="//field[@name='order_line']" position="attributes">
```
Comment (nikul-serpentcs) [inline purchase_order_variant_mgmt/views/purchase_order_view.xml:35]: same here

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: purchase_order_variant_mgmt/tests/test_purchase_order_variant_mgmt.py (hunk 3 above)
Comment (simahawk) [reply purchase_order_variant_mgmt/tests/test_purchase_order_variant_mgmt.py:13]: this is not needed and in any case it's used 2 times

File: purchase_order_variant_mgmt/views/purchase_order_view.xml (hunk 4 above)
Comment (simahawk) [reply purchase_order_variant_mgmt/views/purchase_order_view.xml:10]: not needed

File: purchase_order_variant_mgmt/views/purchase_order_view.xml (hunk 5 above)
Comment (simahawk) [reply purchase_order_variant_mgmt/views/purchase_order_view.xml:35]: same :)

File: purchase_order_variant_mgmt/tests/test_purchase_order_variant_mgmt.py (hunk 3 above)
Comment (nikul-serpentcs) [reply purchase_order_variant_mgmt/tests/test_purchase_order_variant_mgmt.py:13]: Yes It's used 2 times, the second one is [here](https://github.com/OCA/purchase-workflow/pull/594/files/3fc8581da3cb778914ca2f10c9eb18d936637e35#diff-491a9e96ae4c31f244a34367b7182085R20)

File: purchase_order_variant_mgmt/views/purchase_order_view.xml (hunk 4 above)
Comment (nikul-serpentcs) [reply purchase_order_variant_mgmt/views/purchase_order_view.xml:10]: Yes, But You can get `<field>` then why use `xpath`. 

First priority you can use `<field>` then after  `xpath`.

**Comment by pedrobaeza** [comment]
Why is this module being migrated here instead of OCA/product-variant repository? There's also another PR for this module: https://github.com/OCA/product-variant/pull/109

**Comment by grindtildeath** [comment]
@arkostyuk @pedrobaeza I tested the other migration and seems to work, so I suppose we should close this one.

**Comment by pedrobaeza** [comment]
Thanks for the confirmation. Closing.

**Pull request closed without merging**